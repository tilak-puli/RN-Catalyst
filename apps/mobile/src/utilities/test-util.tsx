import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {SpinnerProvider} from '../context/spinner/spinnerContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <QueryClientProvider client={query}>
    <SpinnerProvider>{children}</SpinnerProvider>
  </QueryClientProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
