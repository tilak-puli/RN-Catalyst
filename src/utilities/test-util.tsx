import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '@rneui/themed';
import {SpinnerProvider} from '../context/spinner/spinnerContext';
import {GlobalContextProvider} from '../context/global/globalContext';
import theme from '../styles/theme';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <GlobalContextProvider>
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={query}>
        <SpinnerProvider>{children}</SpinnerProvider>
      </QueryClientProvider>
    </ThemeProvider>
  </GlobalContextProvider>
);

export * from '@testing-library/react-native';
export {customRender as render};
