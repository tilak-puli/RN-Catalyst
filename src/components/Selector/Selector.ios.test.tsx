import {useState} from 'react';
import {ItemValue} from '@react-native-picker/picker/typings/Picker';
import {act, fireEvent, render, screen} from '../../utilities/test-util';
import SelectorIos from './Selector.ios';

const options = [
  {label: 'Option 1', value: 'option1'},
  {label: 'Option 2', value: 'option2'},
];
const mockOptionToSelectIndex = options.length - 1;

jest.mock('react-native/Libraries/ActionSheetIOS/ActionSheetIOS', () => ({
  showActionSheetWithOptions: jest.fn((items, callback) =>
    callback(mockOptionToSelectIndex + 1),
  ), // selecting second option in list
}));

const testId = 'jest-selector';
const ExampleIosSelector = () => {
  const [value, setValue] = useState('' as ItemValue);
  return (
    <SelectorIos
      testID={testId}
      options={options}
      selectedValue={value}
      onValueChange={val => setValue(val)}
    />
  );
};

describe('[Component] - [Selector - iOS]', () => {
  test('should render selector component for ios', () => {
    const select = render(<SelectorIos options={options} />);
    expect(select).toMatchSnapshot();
  });

  test('should show correct number of items in the selector', async () => {
    render(<ExampleIosSelector />);
    const selector = screen.getByTestId(testId);
    const selectorProps = selector.props;
    const selectorItems = selectorProps.children;
    expect(selectorItems.length).toBe(options.length);
  });

  test('should call onValueChange method when one of the option is selected', async () => {
    const selectedOption = options[mockOptionToSelectIndex];

    render(<ExampleIosSelector />);
    const selector = screen.getByTestId(testId);
    act(() => {
      fireEvent(selector, 'onPress');
    });
    expect(screen.getByTestId('selector-selected-value').props.children).toBe(
      selectedOption.label,
    );
  });
});
