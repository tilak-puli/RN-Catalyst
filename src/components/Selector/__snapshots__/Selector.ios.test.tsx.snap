// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - [Selector - iOS] should render selector component for ios 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#eaeaea",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 43,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 14,
      }
    }
    testID="selector-selected-value"
  />
</View>
`;
